test/verify/check-networkmanager-wireguard

```diff
@@ -0,0 +1,178 @@
+        b.set_input_text("#network-wireguard-settings-endpoint-peer-0", f"192.168.100.12:{m2_port}")
+        b.set_input_text("#network-wireguard-settings-allowedips-peer-0", m2_ip4)
+        b.click("button:contains('Add peer')")
+        b.wait_visible("#network-wireguard-settings-peer-1")
+        b.set_input_text("#network-wireguard-settings-publickey-peer-1", m2_pubkey)
+        b.set_input_text("#network-wireguard-settings-endpoint-peer-1", f"192.168.100.12:{m2_port}")
+        b.set_input_text("#network-wireguard-settings-allowedips-peer-1", m2_ip4)
+        b.click("button#network-wireguard-settings-btn-close-peer-0")
+        b.wait_not_present("#network-wireguard-settings-peer-1")
+
+        b.click("button:contains('Save')")
+        b.wait_not_present("#network-wireguard-settings-dialog")
+        b.wait_in_text(f"#networking-interfaces th:contains('{iface_name}') + td", f"1.2.3.4/32, {m1_ip4}/24")
+
+        # endpoint and port is not necessary for a peer if that peer estalishes the connectio first (i.e. the client)
+        m2.execute(f"wg set wg0 peer {m1_pubkey} allowed-ips {m1_ip4}/32")
+
+        # check connection over ipv4
+        try:
+            m1.execute(f"ping -c 5 {m2_ip4}")
+        except (subprocess.CalledProcessError, testlib.Error):
+            m1.execute("ip a >&2; ip route >&2; nmcli c >&2; wg >&2")
+            m2.execute("ip a >&2; ip route >&2; nmcli c >&2; wg >&2")
+            raise
+
+        # check connection over ipv6
+        m2.execute(f"ip addr add {m2_ip6}/64 dev wg0")
+        m2.execute(f"wg set wg0 peer {m1_pubkey} allowed-ips {m1_ip4}/32,{m1_ip6}")
+
+        b.click(f"#networking-interfaces button:contains('{iface_name}')")
+        b.click("#networking-edit-wg")
+        b.wait_visible("#network-wireguard-settings-dialog")
+        b.set_input_text("#network-wireguard-settings-allowedips-peer-0", f"{m2_ip4}/32,{m2_ip6}")
+        b.click("button:contains('Save')")
+        b.wait_not_present("#network-wireguard-settings-dialog")
+
+        b.click("#networking-edit-ipv6")
+        b.wait_visible("#network-ip-settings-dialog")
+        b.select_from_dropdown("#network-ip-settings-select-method", "manual")
+        b.set_input_text("#network-ip-settings-address-0", m1_ip6)
+        b.set_input_text("#network-ip-settings-netmask-0", "64")
+        b.set_input_text("#network-ip-settings-gateway-0", "::")
+        b.click("button:contains('Save')")
+        b.wait_not_present("#network-ip-settings-dialog")
+        self.wait_for_iface_setting("IPv6", "Address 2001:0:0:0:0:0:0:1/64")
+
+        m1.execute(f"until ip a show dev {iface_name} | grep -q 'inet6 {m1_ip6}/64 scope global'; do sleep 0.3; done", timeout=10)
+
+        try:
+            m1.execute(f"ping -6 -c 5 {m2_ip6}")
+        except (subprocess.CalledProcessError, testlib.Error):
+            m1.execute("ip a >&2; ip -6 route >&2; nmcli c >&2; wg >&2")
+            m2.execute("ip a >&2; ip -6 route >&2; nmcli c >&2; wg >&2")
```

Comment:
`nmcli c` isn't good enough here to show the allowed addresses. Perhaps `nmcli c show con-wg0` and `nmcli d show wg0` help? It would be interesting to compare what NM and wg think about which addresses are allowed -- i.e. did the setting from the dialog already fail? Or does NM think it's got an IPv6 address, but didn't write that to wg correctly?

Reply:
With this debug output it works better, and is also easier to read:
```diff
--- test/verify/check-networkmanager-wireguard
+++ test/verify/check-networkmanager-wireguard
@@ -1,6 +1,7 @@
 #!/usr/bin/python3 -cimport os, sys; os.execv(os.path.dirname(sys.argv[1]) + "/../common/pywrap", sys.argv)
 
 import subprocess
+import sys
 
 import netlib
 import packagelib
@@ -150,8 +151,10 @@ class TestWireGuard(packagelib.PackageCase, netlib.NetworkCase):
         try:
             m1.execute(f"ping -6 -c 5 {m2_ip6}")
         except (subprocess.CalledProcessError, testlib.Error):
-            m1.execute("ip a >&2; ip -6 route >&2; nmcli c >&2; wg >&2")
-            m2.execute("ip a >&2; ip -6 route >&2; nmcli c >&2; wg >&2")
+            print("-------- status on m1 ----------", file=sys.stderr)
+            m1.execute("set -x; ip a >&2; ip -6 route >&2; nmcli c >&2; nmcli c show con-wg0 >&2; nmcli d show wg0 >&2; wg >&2")
+            print("-------- status on m2 ----------", file=sys.stderr)
+            m2.execute("set -x; ip a >&2; ip -6 route >&2; nmcli c >&2; wg >&2")
             raise
 
         # install wireguard-tools from the install dialog
```

Unfortunately, `nmcli con show con-wg0` does not show the allowed IPs *at all*, only these:
```
wireguard.private-key:                  <hidden>    
wireguard.private-key-flags:            0 (none)    
wireguard.listen-port:                  51820    
wireguard.fwmark:                       0x0    
wireguard.peer-routes:                  yes    
wireguard.mtu:                          0    
wireguard.ip4-auto-default-route:       -1 (default)    
wireguard.ip6-auto-default-route:       -1 (default)    
```
So it's still unclear at which point setting the allowed IPv6 address failed.

Reply:
Ah! But it's part of the configuration:
```
# grep allowed /etc/NetworkManager/system-connections/con-wg0.nmconnection 
allowed-ips=10.0.0.2;
```

Maybe this is relevant?
```
# journalctl -b | grep 2001
NetworkManager[644]: <warn>  [1691040473.2828] platform: (wg0) route-sync: failure to add IPv6 route: type unicast 2001::2/128 dev 5 metric 50 mss 0 rt-src user: Permission denied
```

That feels more and more like a bug between NM and the kernel. @subhoghoshX can you please create a command line reproducer for it, with `nmcli`? If you can't reproduce that, I'm happy to try here. Then we can submit that as a NM bug report and add a naughty pattern for it.

Reply:
Thanks for looking into it. Unfortunately even after running 7 parallel tests and YT I can't seem to reproduce it. Going further it only breaks on `self.login_and_go("/network")` mostly :-(

Reply:
I restructured this a bit to ensure that setting the allowed IPs actually ends up in `wg show`. This both makes the error easier to debug, and also avoids the race of trying to modify the IPv6 configuration twice asynchronously. It's sort of avoiding the bug, but if it still fails now, we can write an easier naughty pattern.